src/frontend/src/pages/PIADetailPage/index.tsx

```diff
@@ -0,0 +1,187 @@
+import {
+  IPIAIntake,
+  IPIAIntakeResponse,
+} from '../../types/interfaces/pia-intake.interface';
+import { faArrowDown } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { dateToString } from '../../utils/date';
+import messages from './messages';
+
+import { useNavigate, useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { HttpRequest } from '../../utils/http-request.util';
+import { API_ROUTES } from '../../constant/apiRoutes';
+import { routes } from '../../constant/routes';
+import Alert from '../../components/common/Alert';
+import MDEditor from '@uiw/react-md-editor';
+
+const PIADetailPage = () => {
+  // https://github.com/microsoft/TypeScript/issues/48949
+  // workaround
+  const win: Window = window;
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const [pia, setPia] = useState<any>({});
+  const [fetchPiaError, setFetchPiaError] = useState('');
+  useEffect(() => {
+    (async () => {
+      try {
+        // Actually perform fetch
+        const result = (
+          await HttpRequest.get<IPIAIntakeResponse>(
+            API_ROUTES.GET_PIA_INTAKE.replace(':id', `${id}`),
+          )
+        ).data;
+        setPia(result);
+      } catch (e) {
+        if (e instanceof Error && e.cause instanceof Error) {
```

Comment:
I don't think this is right. e can be instance of error, e.cause is not necessarily instance of error:

[Description](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause#description)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause
The value of cause can be of any type. You should not make assumptions that the error you caught has an Error as its cause, in the same way that you cannot be sure the variable bound in the catch statement is an Error either. The "Providing structured data as the error cause" example below shows a case where a non-error is deliberately provided as the cause.

Reply:
For the exact same reason, `e.cause` is explicitly created as custom error in the http-request.util.ts;

The idea was to initially have the error status mentioned in the cause, but as Bing found out earlier, it was throwing an error while building, not necessarily at compile time.

check this thread above - https://github.com/bcgov/cirmo-dpia/pull/710#discussion_r1061069621

Reply:
Ah! Typescript